front/lib/auth.ts

```diff
@@ -22,7 +22,7 @@ const {
   NODE_ENV,
 } = process.env;
 
-export type RoleType = "admin" | "builder" | "user" | "none";
+export type RoleType = "admin" | "builder" | "user" | "none" | "system";
```

Comment:
I see the motivation will submit the review right now but need to think more about it as this is touching a pretty sensible part of the codebase :grimacing: